manifests/backup/restore-gcs.yaml

```diff
@@ -0,0 +1,19 @@
+---
+apiVersion: pingcap.com/v1alpha1
+kind: Restore
+metadata:
+  name: demo2-restore
+  namespace: test2
+spec:
+  from:
+    host: 10.0.0.2
+    port: 4000
+    user: root
+    secretName: backup-demo2-tidb-secret
+  gcs:
+    projectId: <your-project-id>
+    secretName: gcs-secret
+  storageType: gcs
```

Comment:
Putting `storageType`, `storageClassName` and `storageSize` together is confusing.

Is the PV a temporary storage? Maybe we can rename either the storage backend or the temporary PV.

Also, I think the backup path is specific to each storage backend, it's better to put under the corresponding storage backend.

Reply:
How about this?
```yaml
spec:
  from: # storage backend info for backup file
    type: gcs
    gcs:
      projectId: <your-project-id>
      secretName: gcs-secret
      path: gcs://test1-demo1/backup-2019-11-11T16:06:05Z.tgz
  to: # downstream tidb cluster info
    host: 10.0.0.2
    port: 4000
    user: root
    secretName: backup-demo2-tidb-secret
  tempStorage: # use temporarily to download the backup file
    storageClassName: local-storage
    size: 1Gi
```
/cc @DanielZhangQD @shuijing198799

Reply:
I agree putting `storageType`, `storageClassName` and `storageSize` together is confusing. These three fields all start with the word storage, which makes people think they have any relationship, but there is no relationship between them. Therefore I intend to remove `storageType` field, here is the related issue #768.  I prefer the revised definition below.

```yaml
spec:
  to:
    host: 10.0.0.2
    port: 4000
    user: root
    secretName: backup-demo2-tidb-secret
  gcs:
    projectId: <your-project-id>
    secretName: gcs-secret
    # "<bucket-name>/<path-to-backup-file>"
    path: test1-demo1/backup-2019-11-11T16:06:05Z.tgz
  storageClassName: local-storage
  storageSize: 1Gi
```
What do you think? @DanielZhangQD @shuijing198799

Reply:
use `from` and `to` is more clear for users.
I agree to remove the `storageType` attribute, `gcs:` indicate the type already.

```
spec:
  storageClassName: local-storage
  storageSize: 1Gi
  from:
    gcs:
      projectId: <your-project-id>
      secretName: gcs-secret
      path: gcs://test1-demo1/backup-2019-11-11T16:06:05Z.tgz
  to:
    host: 10.0.0.2
    port: 4000
    user: root
    secretName: backup-demo2-tidb-secret
```

Reply:
I think that defining a remote storage configuration(here is `gcs`) in restore has already indicated that the backup data must come from here, and there is no need to use `from` to indicate this relationship. In addition, we should ensure that the `spec` is as simple as possible to reduce the cost of use for users.

Reply:
There are storage class and size defined, users may think the backup is in the storage defined with `storageClassName` and `storageSize`. Adding `to` makes it less confuse and more straight forward and also simple enough.

Reply:
I think no need to add `from/to`, users know they are doing `backup` or `restore`, so even if no `from/to` indication, they know that the tidb cluster are used for what.
For the `storageClassName: local-storage  storageSize: 1Gi`, I prefer to make it clear in the doc that they are used for temporary storage whether to put them under `tempStorage` or not.
```
spec:
  cluster or tidb:
    host: 10.0.0.2
    port: 4000
    user: root
    secretName: backup-demo2-tidb-secret
  gcs:
    projectId: <your-project-id>
    secretName: gcs-secret
    # "<bucket-name>/<path-to-backup-file>"
    path: test1-demo1/backup-2019-11-11T16:06:05Z.tgz
  storageClassName: local-storage
  storageSize: 1Gi
```

Reply:
What if users provide multiple cloud storage configurations if we omit the `storageType` field?